status/k8s.go

```diff
@@ -311,99 +321,168 @@ retry:
+					}
+				}
 
-	disabled, err = k.deploymentStatus(ctx, status, defaults.HubbleUIDeploymentName)
-	status.SetDisabled(defaults.HubbleUIDeploymentName, defaults.HubbleUIDeploymentName, disabled)
-	if err != nil {
-		if _, ok := status.PodState[defaults.HubbleUIDeploymentName]; !ok {
-			status.AddAggregatedWarning(defaults.HubbleUIDeploymentName, defaults.HubbleUIDeploymentName, fmt.Errorf("hubble ui is not deployed"))
-		} else {
-			status.AddAggregatedError(defaults.HubbleUIDeploymentName, defaults.HubbleUIDeploymentName, err)
-			status.CollectionError(err)
-		}
+				// pod status for relay is only validated if the deployment exists
+				if _, ok := status.PodState[defaults.ClusterMeshDeploymentName]; ok {
+					err = k.podStatus(ctx, status, defaults.ClusterMeshDeploymentName, "k8s-app=clustermesh-apiserver", nil)
+					if err != nil {
+						status.CollectionError(err)
+					}
+				}
+
+				return nil
+			},
+		},
+		statusTask{
+			name: defaults.HubbleUIDeploymentName,
+			task: func(_ context.Context) error {
+				err := k.podCount(ctx, status)
+				if err != nil {
+					status.CollectionError(err)
+				}
+
+				return nil
+			},
+		},
 	}
 
-	// pod status for UI is only validated if the deployment exists
-	if _, ok := status.PodState[defaults.HubbleUIDeploymentName]; ok {
-		err = k.podStatus(ctx, status, defaults.HubbleUIDeploymentName, "k8s-app=hubble-ui", nil)
-		if err != nil {
-			status.CollectionError(err)
+	var mutex sync.Mutex
+	err := k.podStatus(ctx, status, defaults.AgentDaemonSetName, "k8s-app=cilium", func(ctx context.Context, status *Status, name string, pod *corev1.Pod) {
+		if pod.Status.Phase == corev1.PodRunning {
+			tasks = append(tasks, statusTask{
+				name: name,
+				task: func(_ context.Context) error {
+					s, err := k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
+					status.parseStatusResponse(defaults.AgentDaemonSetName, pod.Name, s, err)
+					mutex.Lock()
+					defer mutex.Unlock()
+					status.CiliumStatus[name] = s
+
+					return nil
+				},
+			})
 		}
-	}
+	})
 
-	disabled, err = k.deploymentStatus(ctx, status, defaults.ClusterMeshDeploymentName)
-	status.SetDisabled(defaults.ClusterMeshDeploymentName, defaults.ClusterMeshDeploymentName, disabled)
 	if err != nil {
-		if _, ok := status.PodState[defaults.ClusterMeshDeploymentName]; !ok {
-			status.AddAggregatedWarning(defaults.ClusterMeshDeploymentName, defaults.ClusterMeshDeploymentName, fmt.Errorf("clustermesh is not deployed"))
-		} else {
-			status.AddAggregatedError(defaults.ClusterMeshDeploymentName, defaults.ClusterMeshDeploymentName, err)
-			status.CollectionError(err)
-		}
+		status.CollectionError(err)
 	}
 
-	// pod status for relay is only validated if the deployment exists
-	if _, ok := status.PodState[defaults.ClusterMeshDeploymentName]; ok {
-		err = k.podStatus(ctx, status, defaults.ClusterMeshDeploymentName, "k8s-app=clustermesh-apiserver", nil)
-		if err != nil {
+	wp := workerpool.New(k.params.WorkerCount)
```

Comment:
We need to add a safeguard here: the number of workers must be > 0 or `workerpool.New` panics.